### ISSUE TITLE
Weekly/ Monthly digest for publiclab.org

### ISSUE THREAD

**Issue by ananyo2012** [publiclab/plots2#1119]
Just thought of moving this to a separate issue Based on tag subscriptions we can send customized digest. 
Digest can include research notes as well as questions.

Rough steps that need to be followed
- [ ] Suggest a possible design for the email template
- [ ] Add a method in [Subscription mailer](https://github.com/publiclab/plots2/blob/master/app/mailers/subscription_mailer.rb) for sending digests
- [ ] Defining a method in [User Model](https://github.com/publiclab/plots2/blob/master/app/models/user.rb) to call the Subscription Mailer method since subscriptions  will be based on tag preferences 
- [ ] Think of a way of sending email digests. One way to do this is using a cronjob. We can use the [whenever](https://github.com/javan/whenever) gem. For that we have to create a Rake task as well.

**Comment by jywarren** [comment]
This may need to be broken into smaller parts to be effectively solved; adding `break-me-up` and we also need:

* [ ] a way to choose who receives email digests. We could start with just sending to all admins, to test the system out before turning it on for a larger group.

**Comment by ananyo2012** [comment]
@jywarren Yes please list way to break up the issue. This is labelled as a Google Code-in task so we need to make it as clear to students as possible. If it isn't taken by by end of the contest we can take it up ourselves as well.

**Comment by jywarren** [comment]
I think this needs some thinking through -- but what about if:

* [ ] subscriptions (such as `TagSelection`s, which along with `NodeSelection`s, are the primary means of subscribing) could have a type attribute -- like `type = immediate` or `type = weekly` or `type = daily`
* [ ] we developed a method like `DrupalTag.daily_digest_recipients` (or maybe `TagSelection.digest_recipients(period)`) which collected all users who have subscribed to a tag which has been used in the specified period (like `24.hours` using RoR time syntax)
* [ ] we made a `User.subscribed_content(period)` which collected all the nodes answers, and comments made in the specified period
* [ ] figure out how to use [Active Job](http://edgeguides.rubyonrails.org/active_job_basics.html) in Rails 3 (keeping in mind we'll move to Rails 4 before too long, so check if it changes a lot) and develop a daily/weekly job to collect subscribed users and generate a text list that can be mailed. 

If we do this right, we can write tests against all of these steps before actually writing a mailer or testing by email, and we can ensure this'll work without having to do risky email testing on production!

**Comment by jywarren** [comment]
Worked on breaking this up in https://github.com/publiclab/plots2/issues/2027 and we'll need some of this in there too -- specifically the stuff in my last comment about finding the people who should get notified at a specific time! I think we may want to switch to a profile tag like `digest:daily` or `digest:weekly` and send to everyone once per day using that... not sure.

**Comment by steviepubliclab** [comment]
This type of curated information could also be useful for the https://publiclab.org/wiki/requests page.

**Comment by ebarry** [comment]
There seems to be language on the /subscriptions page that is both leftover from GoogleGroups and aspirational-but-not-yet-achieved until the features in this issue are created: 

<img width="748" alt="screen shot 2018-01-24 at 4 34 45 pm" src="https://user-images.githubusercontent.com/161439/35360616-58375acc-012c-11e8-9ef5-bd1445a31f6c.png">

The actual text reads: 
>Subscriptions let you keep up to date with topics on Public Lab. You can choose to receive them by email or in digest form.

We should remove the second sentence and just keep the first until these features exist. 

Am i wrong?

**Comment by jywarren** [comment]
Great -- however, we can change that to `or view them in digest form` linked to `/subscriptions/digest` as soon as https://github.com/publiclab/plots2/pull/2058 is complete -- that implements the initial version of the weekly digest (although it's not emailed yet -- just displayed!)

**Comment by jywarren** [comment]
This is really getting broken up into pieces and solved!!! :-) 👍 🎈 💥

**Comment by visheshruparelia** [comment]
I'd like to work on this for GSOC2018. Can anyone guide me further on this?

**Comment by steviepubliclab** [comment]
Hi there,
All the planning and organizing for GSoC will be on this page
<https://publiclab.org/wiki/gsoc>. Looking forward to seeing your ideas.
-Stevie

On Wed, Feb 14, 2018 at 2:21 AM, Vishesh Ruparelia <notifications@github.com

**Comment by jywarren** [comment]
Hi, @visheshruparelia -- please take a look through https://publiclab.github.io/community-toolbox/#r=all -- we'd love to have your help with one of those! Or if they're all taken, perhaps a https://github.com/publiclab/plots2/labels/help-wanted

however, if @Gauravano has an idea of where you could help with this one, maybe!

**Comment by ebarry** [comment]
Great to see that https://publiclab.org/subscriptions/digest exists and is working! 
What improvements should come next?

**Comment by jywarren** [comment]
https://github.com/publiclab/plots2/issues/2378 is a good next step!

**Comment by ViditChitkara** [comment]
Hi everyone,
Thinking of using resque as the queuing backend for this (maybe we'd require capistrano for this work on production), @icarito could you please give your thoughts on this? Please look through https://publiclab.org/notes/vidit/02-16-2018/gsoc-18-email-notification-overhaul#c18986 
@jywarren , @ebarry , @steviepubliclab , @ananyo2012 I'd love to have your reviews too.

**Comment by jywarren** [comment]
Hi, i'll try to take a look tomorrow, thanks! Just returned from a long flight (dubai!). I'm going to add a few items to the top of this issue, though, including:

* ability to change timeframe of `/subscriptions/digest` to `weekly` or `monthly` like [Github Pulse](https://github.com/publiclab/plots2/pulse/weekly) has
* should we try to merge some of the back-end methods (Tag or Node models) with those we use in the Tag pages, where we're also trying to enable time slices? Or the Stats pages?

**Comment by jywarren** [comment]
@ViditChitkara perhaps we can try to summarize and consolidate as much as we can from this into your new planning issue? Or consolidate more here? Either way!

**Comment by ViditChitkara** [comment]
I am comfortable with both!! I guess we can discuss things to be done on the planning issue.
